src/main/java/sirius/kernel/commons/Value.java

```diff
@@ -539,31 +539,36 @@ public Value first() {
 
     @SuppressWarnings("unchecked")
     private <T> T continueCoerceWithDateTypes(Class<T> targetClazz, T defaultValue) {
-        if (LocalDate.class.equals(targetClazz)) {
-            if (is(TemporalAccessor.class, Calendar.class, Date.class, java.sql.Date.class, Timestamp.class)) {
-                return (T) asLocalDate((LocalDate) defaultValue);
-            }
-        }
-        if (LocalDateTime.class.equals(targetClazz)) {
-            if (is(TemporalAccessor.class, Calendar.class, Date.class, java.sql.Date.class, Timestamp.class)) {
-                return (T) asLocalDateTime((LocalDateTime) defaultValue);
-            }
-        }
-        if (ZonedDateTime.class.equals(targetClazz)) {
-            if (is(TemporalAccessor.class, Calendar.class, Date.class, java.sql.Date.class, Timestamp.class)) {
-                return (T) asZonedDateTime((ZonedDateTime) defaultValue);
-            }
-        }
-        if (LocalTime.class.equals(targetClazz) && data instanceof TemporalAccessor) {
-            if (is(TemporalAccessor.class,
-                   Calendar.class,
-                   Date.class,
-                   java.sql.Date.class,
-                   Timestamp.class,
-                   Time.class)) {
-                return (T) asLocalTime((LocalTime) defaultValue);
-            }
+        if (LocalDate.class.equals(targetClazz) && is(TemporalAccessor.class,
+                                                      Calendar.class,
+                                                      Date.class,
+                                                      java.sql.Date.class,
+                                                      Timestamp.class)) {
+            return (T) asLocalDate((LocalDate) defaultValue);
+        }
+        if (LocalDateTime.class.equals(targetClazz) && is(TemporalAccessor.class,
+                                                          Calendar.class,
+                                                          Date.class,
+                                                          java.sql.Date.class,
+                                                          Timestamp.class)) {
+            return (T) asLocalDateTime((LocalDateTime) defaultValue);
+        }
+        if (ZonedDateTime.class.equals(targetClazz) && is(TemporalAccessor.class,
+                                                          Calendar.class,
+                                                          Date.class,
+                                                          java.sql.Date.class,
+                                                          Timestamp.class)) {
+            return (T) asZonedDateTime((ZonedDateTime) defaultValue);
+        }
+        if (LocalTime.class.equals(targetClazz) && data instanceof TemporalAccessor && is(TemporalAccessor.class,
```

Comment:
sorry if I am missing something but how could data be assignable to e.g. Timestamp or Calendar when its an instance of TemporalAccessor?

Reply:
Merge readded the wrong instanceof check